tests/test_synced_collection.py

```diff
@@ -0,0 +1,295 @@
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import uuid
+from tempfile import TemporaryDirectory
+
+from signac.core.collection_api import SyncedDict
+from signac.core.collection_api import SyncedList
+from signac.errors import InvalidKeyError
+from signac.errors import KeyTypeError
+
+
+def testdata():
+    return str(uuid.uuid4())
+
+
+class TestSyncedCollectionBase():
+
+    @pytest.fixture(autouse=True)
+    def setUp(self, request):
+        self._tmp_dir = TemporaryDirectory(prefix='jsondict_')
+        request.addfinalizer(self._tmp_dir.cleanup)
+
+    def get_testdata(self):
+        return str(uuid.uuid4())
+
+
+class TestSyncedDict(TestSyncedCollectionBase):
+
+    def get_synced_dict(self, data=None):
+        return SyncedDict(data)
```

Comment:
If you move this to `TestSyncedCollectionBase` and redefine it a
```suggestion
    def get_synced_collection(self, data=None):
        return self._type(data)
```
or similar, you might be able to standardize some tests (like `test_init`) across all `SyncedCollection`s and not have to duplicate them.